pkg/proxy/server.go

```diff
@@ -1067,7 +1067,7 @@ func (s *Server) Proxy() {
 }
 
 func (s *Server) sendConnectErrorMsg(err error) {
-	msg := fmt.Sprintf("%s error: %s", s.connOpts.ConnectMsg(),
+	msg := fmt.Sprintf("%s error: %s", s.connOpts.ConnectMsg(s.jmsService),
 		s.ConvertErrorToReadableMsg(err))
 	utils.IgnoreErrWriteString(s.UserConn, msg)
 	utils.IgnoreErrWriteString(s.UserConn, utils.CharNewLine)
```

Comment:
I suggest checking the above code thoroughly for any irregularities, issues, or optimizations. I can also help with further debugging or refinement if needed.